src/containers/audio-rewards-page/ChallengeRewardsTile.tsx

```diff
@@ -0,0 +1,167 @@
+// @ts-nocheck
+import React, { ReactNode, useEffect } from 'react'
+
+import cn from 'classnames'
+import { useDispatch, useSelector } from 'react-redux'
+
+import LoadingSpinner from 'components/loading-spinner/LoadingSpinner'
+import { useRemoteVar } from 'containers/remote-config/hooks'
+import { useSetVisibility } from 'hooks/useModalState'
+import { useWithMobileStyle } from 'hooks/useWithMobileStyle'
+import { StringKeys } from 'services/remote-config'
+import fillString from 'utils/fillString'
+
+import styles from './RewardsTile.module.css'
+import ButtonWithArrow from './components/ButtonWithArrow'
+import { Tile } from './components/ExplainerTile'
+import ProgressBar from './components/ProgressBar'
+import { challengeRewardsConfig } from './config'
+import {
+  fetchUserChallenges,
+  ChallengeRewardsModalType,
+  setChallengeRewardsModalType,
+  getUserChallenges,
+  getUserChallengesLoading
+} from './store/slice'
+import { ChallengeRewardID } from './types'
+
+const TILE_TITLE = '$AUDIO REWARDS'
+const TILE_DESC1 = 'Complete tasks to earn $AUDIO tokens!'
+const TILE_DESC2 =
+  'Opportunities to earn $AUDIO will change, so check back often for more chances to earn!'
+const COMPLETE_LABEL = 'COMPLETE'
+
+type RewardPanelProps = {
+  title: string
+  icon: ReactNode
+  description: string
+  buttonText: string
+  progressLabel: string
+  stepCount: number
+  openModal: (modalType: ChallengeRewardsModalType) => void
+  id: ChallengeRewardID
+}
+
+const RewardPanel = ({
+  id,
+  title,
+  description,
+  buttonText,
+  openModal,
+  progressLabel,
+  icon,
+  stepCount
+}: RewardPanelProps) => {
+  const wm = useWithMobileStyle(styles.mobile)
+  const userChallenges = useSelector(getUserChallenges)
+
+  const openRewardModal = () => openModal(id)
+
+  const challenge = userChallenges.find(
+    userChallenge => userChallenge.challenge_id === id
+  )
+  const currentStepCount = challenge?.current_step_count || 0
```

Comment:
Looking the the UserChallenge type, current_step_count is not optional

Reply:
This is for the case where the user challenge is not created. should default to zero bc the user hasn't started the challenge